addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java

```diff
@@ -59,12 +74,29 @@ public void init() throws IOException {
             dtoState = LEDStateDTO.valueOf(s.state, s.program, s.programSpeed, s.red, s.green, s.blue, s.white,
                     s.white2);
             power = (s.state & 0x01) != 0;
-            currentTargetState = InternalLedState.fromRGBW(s.red, s.green, s.blue, s.white, s.white2);
+            currentFaderState = currentState = InternalLedState.fromRGBW(s.red, s.green, s.blue, s.white, s.white2);
         } catch (IOException ignored) {
         }
     }
 
     @Override
+    public void dispose() {
+        waiterFuture.shutdown();
```

Comment:
The dispose is also used in other cases for example when re-initializing if the configuration has changed. So after the dispose the initialize is called and the handler will be re-used. However in this case both executorservices will already be shutdown and will accept no further tasks.

An example of problems with this can be seen in this PR: https://github.com/openhab/openhab2-addons/pull/2437